deeppavlov/skills/dsl_skill/dsl_skill.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2019 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta
+from collections import defaultdict
+from functools import partial
+from itertools import zip_longest, starmap
+from typing import List, Optional, Dict, Callable, Tuple
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.skills.dsl_skill.context import UserContext
+from deeppavlov.skills.dsl_skill.handlers import Handler, RegexHandler
+from deeppavlov.skills.dsl_skill.utils import SkillResponse, UserId
+
+
+class DSLMeta(ABCMeta):
+    """
+    This metaclass is used for creating a skill. Skill is register by its class name in registry.
+
+    Example:
+
+    .. code:: python
+
+            class ExampleSkill(metaclass=DSLMeta):
```

Comment:
This example is different from the one given in `dsl_skill.rst` and provides different output. In `dsl_skill.rst` we have an example without `state` argument. If I run the example with `state = "greeting"`, I will get `[['Sorry, I do not understand you'], [0.0]]` output. Without `state = "greeting"` I am getting `[['Hello, my friend!'], [1.0]]` output. Please copypaste either here or to the docs the correct example.

Reply:
Done